### DISCUSSION TITLE
How to get started.

### DISCUSSION THREAD

**Discussion by mattjoyce** [QutEcoacoustics/audio-analysis#561] [General]
As an amateur field recording, I have have collected decent quality recordings from frog ponds, and I would like to figure out a way to identify which frog calls have been captured.  My goal is to automate the identification of known species.

When I listen and  view the recordings in Audacity, I can identify the sounds that are distinct, it those sounds I want to detect.
After googling and reading , I found this project, which seems to be exactly the goal.

I have been reading documentation and config files, but I'm must be missing something key.
I thought I could take an existing frog yaml, something similar to what I need, make a copy and adjust some parameters.
when I run audio2csv with my config, it says there's no analysis identifier.

ok,, so `ap AnalysesAvailable`
why are there specific species in that list where do they come from if they are not config files, how do I make my own?

I'm a bit stuck.  I'm happy to put in the time to learn and iterate a recogniser, but I'm not sure how to start.

Reading the doc on Generic Recogniser seemed promising (and super interesting).
If I understood correctly, I make a config file, and use 'Ecosounds.GenericRecognizer' as the analysis identifier.

Edit:
ok, I tried that, made a copy Ecosounds.GenericRecognizer.yml, trimmed the file to have a single profile Oscillator profile , broadened the parameters to hopefully find something.  
Running this produces the error.
Fatal error:
  Configuration exception in profile Test1:
        - DecibelThresholds: must be set and be not null in the config file

I can find no information on this parameter.
I have the singular DecibelThreshold set as show in the example files.

[Ecosounds.GenericRecognizer.test.yml.txt](https://github.com/QutEcoacoustics/audio-analysis/files/13305512/Ecosounds.GenericRecognizer.test.yml.txt)

**Comment by atruskie** [comment]
Hi @mattjoyce,

First of all, a disclaimer. AP is generally not being developed much anymore and other the person responsible for the generic recognizers has retired. AP might not be your best choice.

I assume you've read this documentation: https://ap.qut.ecoacoustics.info/guides/generic_recognizers ?

Second, all of our frog recognizers were developed using an older technology. They're not generic recognizers and they also don't work very well. Once again, the people responsible for maintaining them have moved on.

But you can take an existing config file, like the one you picked and adapt it to your needs. Any of the ones that have  the `PostProcessing` section are using the generic recognizer code. e.g. https://github.com/QutEcoacoustics/audio-analysis/blob/master/src/AnalysisConfigFiles/RecognizerConfigFiles/Towsey.NinoxBoobook.yml


For the particular error you mentioned, try setting `DecibelThresholds` as per https://github.com/QutEcoacoustics/audio-analysis/blob/23a0bbf05c5f59638dbb195a7bc613096d64a6b5/src/AnalysisConfigFiles/RecognizerConfigFiles/Towsey.NinoxBoobook.yml#L27-L30. At some point it changed from `DecibelThreshold` (a single value) to an array of values.

**Comment by mattjoyce** [comment]
I see, well thank you for your quick response.  I will invest a bit more
time, it's interesting.

On Thu, 9 Nov 2023 at 19:15, Anthony Truskinger ***@***.***>
wrote:

> Hi @mattjoyce <https://github.com/mattjoyce>,
>
> First of all, a disclaimer. AP is generally not being developed much
> anymore and other the person responsible for the generic recognizers has
> retired. AP might not be your best choice.
>
> I assume you've read this documentation:
> https://ap.qut.ecoacoustics.info/guides/generic_recognizers ?
>
> Second, all of our frog recognizers were developed using an older
> technology. They're not generic recognizers and they also don't work very
> well. Once again, the people responsible for maintaining them have moved on.
>
> But you can take an existing config file, like the one you picked and
> adapt it to your needs. Any of the ones that have the PostProcessing
> section are using the generic recognizer code. e.g.
> https://github.com/QutEcoacoustics/audio-analysis/blob/master/src/AnalysisConfigFiles/RecognizerConfigFiles/Towsey.NinoxBoobook.yml
>
> For the particular error you mentioned, try setting DecibelThresholds as
> per
> https://github.com/QutEcoacoustics/audio-analysis/blob/23a0bbf05c5f59638dbb195a7bc613096d64a6b5/src/AnalysisConfigFiles/RecognizerConfigFiles/Towsey.NinoxBoobook.yml#L27-L30.
> At some point it changed from DecibelThreshold (a single value) to an
> array of values.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/QutEcoacoustics/audio-analysis/discussions/561#discussioncomment-7519449>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACECVP4KFOMTNRG764NOC3YDSGKJAVCNFSM6AAAAAA7EB4UIWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TKMJZGQ2DS>
> .
> You are receiving this because you were mentioned.Message ID:
> <QutEcoacoustics/audio-analysis/repo-discussions/561/comments/7519449@
> github.com>
>


-- 
+61 468 335597
*LinkedIn Profile : Matt Joyce
<http://au.linkedin.com/pub/matthew-joyce/6/4b5/6a4/>*
